Create place page

loricof23 opened the pull request:
Make a Place Page based on Heikaku Website and Add Searching Fiture

loricof23 committed:
SCSS Page

loricof23 committed:
Add PlacePage

loricof23 committed:
Index JS

loricof23 committed:
Package-logJson

loricof23 committed:
package-log-json

loricof23 committed:
TestPage

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -0,0 +1,62 @@
+<template>
+  <div class="test-page">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Link to external resource should be put in the `index.html`

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -0,0 +1,62 @@
+    <div class = "searchbar">
+      <center>
+      <input class="text-search" type="text" v-model="search" placeholder="Nama Tempat Yang dicari" @change="searching" >
+      <input class="button" type="submit" value ="cari">
```
You need to decide what event that you're going to used. @change or @click?

Because if @change, submit button is not necessary anymore since the input already handle the method. But if @click, you have to remove @change event from the input, and put @click event in the submit button

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -0,0 +1,62 @@
+      </center>
+    </div>
+    <div class= "box">
+        <div class = "item" v-for="place in places" :key="place.Id">
```
```suggestion
        <div class="item" v-for="place in places" :key="place.Id">
```
I think it is a little weird if you add space between attribute and its value

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -0,0 +1,62 @@
+        <header>
+            <img :src="place.Company.Photo.Url" class="photo">
+            <div class = "partner">{{ place.Company.Name }}</div>
+            <div class = "non-rate" v-if="place.Rating == null"> </div>
```
```suggestion
            <div class = "non-rate" v-if="!place.Rating"> </div>
```

Shorter condition to check that the value is null

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -0,0 +1,62 @@
+  data () {
+    return {
+      places: [],
+      moment: moment,
```
No need to create a new variable for this reason

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -0,0 +1,62 @@
+                   <div class="name">{{ place.Name }} </div>
+                   <div class="city">{{ place.City.Name }} </div>
+                   <div class="hour">{{ moment (place.Times[0].OpenHour).format('LT') }} - {{ moment (place.Times[0].CloseHour).format('LT') }}
+                      <span class="close" v-if="checkHourInfo(place.Times)"> Close Now </span>
```
Your implementation to show the operation hours is not completely right.

In here, you only get the first `OpenHour` and `CloseHour` in that `Times` list. What if the first day in that list is `Monday` but today is `Tuesday`? In addition, the operation hours in those days is different, then your implementation goes wrong because that's not the right operation hours for this day

**Example:** 
Monday: 10.00 - 15:00
Tuesday: 15:00 - 17:00 (Today)

You can see a different here isn't it?

zalhamami replied:
You need to get this day from js and check `TimeId` in the `Times` list. 

`TimeId` 1 is monday
`TimeId` 2 is tuesday
`TimeId` 3 is wednesday
and so on..

zalhamami replied:
Here's an example for this algorithm:

```
currentDate = get today date
currentDay = get day from current date
timeObject = find object in the list Times by checking TimeId must same with this day

currentTime = get current time by moment
openHour = get open hour time by moment from timeObject
closeHour = get close hour time by moment from timeObject

if (currentTime > openHour && currentTime < closeHour) then
  'open'
else
  'close'
```

zalhamami commented on package.json:
```diff
@@ -15,6 +15,7 @@
   },
   "dependencies": {
     "axios": "^0.19.2",
+    "moment": "^2.24.0",
```
The commit message should be `Install moment.js`. Fix it next time ya

zalhamami commented on src/router/index.js:
```diff
@@ -16,6 +16,11 @@ export default new Router({
+      path: '/place',
+      component: PlacePage
     }
+
```
The commit message should be `Add place page routes`

zalhamami commented:
Learn to make clear for commit message, Let's imagine you have a dozens commit, it's gonna be hard for reviewers to review your code

zalhamami committed:
Merge branch 'master' into placePage

loricof23 committed:
Update Code

Move external source to index.html,Changes @changes to @click,changes attribute space,Implementation hours

loricof23 committed:
Update Code

Link rel to index.html

loricof23 committed:
Merge branch 'placePage' of https://github.com/zalhamami/vue-test into placePage

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -32,23 +33,50 @@
+    },
+    dayInfo (times) {
+      return moment(times[0].TimeOfUse.Name).format('dddd')
     },
```
Remove these unnecessary functions

zalhamami commented on src/pages/PlacePage.vue:
```diff
@@ -1,28 +1,29 @@
             <div class = "partner">{{ place.Company.Name }}</div>
-            <div class = "non-rate" v-if="place.Rating == null"> </div>
+            <div class = "non-rate" v-if="!place.Rating"> </div>
             <div class = "rate" v-else> <span class="fa fa-star checked"></span>{{ place.Rating }}/{{ place.RatingCount }} </div>
```
Again, no need to add white space between attribute and its value

loricof23 committed:
Fix space and delete function

loricof23 committed:
Remove Function
